spacy/cli/project/parallel.py

```diff
@@ -0,0 +1,442 @@
+                elif cmd_info.rc > 0:
+                    cmd_info.change_state("failed")
+                else:
+                    cmd_info.change_state("terminated")
+                cmd_info.console_output = cast(str, mess["console_output"])
+            if any(
+                c for c in cmd_infos if c.state.name in ("failed", "terminated", "hung")
+            ):
+                # a command in the group hasn't succeeded, so terminate/cancel the rest
+                for other_cmd_info in (
+                    c for c in cmd_infos if c.state.name == "running"
+                ):
+                    try:
+                        os.kill(cast(int, other_cmd_info.pid), SIGTERM)
+                    except:
+                        # there could be a race condition between the process completing
+                        # and the message arriving to notify about it
+                        pass
+                for other_cmd_info in (
+                    c for c in cmd_infos if c.state.name == "pending"
+                ):
+                    other_cmd_info.change_state("cancelled")
+            if mess["type"] != "keepalive" and DISPLAY_STATUS_TABLE:
+                if first:
+                    first = False
+                else:
+                    # overwrite the existing status table
+                    print("\033[2K\033[F" * (4 + len(cmd_infos)))
+                data = [[c.cmd_name, c.state_repr] for c in cmd_infos]
+                header = ["Command", "Status"]
+                msg.table(data, header=header)
+
+        for cmd_info in (c for c in cmd_infos if c.state.name != "cancelled"):
+            msg.divider(cmd_info.cmd_name)
+            if cmd_info.state.name == "not rerun":
+                msg.info(f"Skipping '{cmd_info.cmd_name}': nothing changed")
+            elif cmd_info.console_output is not None:
+                print(cmd_info.console_output)
+        # Occasionally when something has hung there may still be worker processes to tidy up
+        for hung_worker_process in (wp for wp in worker_processes if wp.is_alive()):
+            hung_worker_process.terminate()
+        process_rcs = [c.rc for c in cmd_infos if c.rc is not None]
+        if len(process_rcs) > 0:
+            group_rc = max(process_rcs)
+            if group_rc > 0:
+                msg.fail("A command in the parallel group failed.")
+                sys.exit(group_rc)
+        if any(c for c in cmd_infos if c.state.name in ("hung", "terminated")):
+            msg.fail("Command(s) in the parallel group hung or were terminated.")
+            sys.exit(-1)
+
+
+def _project_run_parallel_cmd(
+    cmd_info: _ParallelCommandInfo,
+    *,
+    dry: bool,
+    current_dir: str,
+    parallel_group_status_queue: Queue,
+) -> None:
+    """Run a single spaCy projects command as a worker process.
+
+    Communicates with the main process via queue messages that consist of
+    dictionaries whose type is determined by the entry 'type'. Possible
```

Comment:
Nitpick: I'm not sure if `type` is ideal here due to its connotation to variable types. Maybe `status`? Feel free to ignore though.

Reply:
I've changed the name to `mess_type`.